homeassistant/helpers/script.py

```diff
@@ -450,7 +453,7 @@ async def async_run(self) -> ScriptRunResult | None:
             script_stack.pop()
             self._finish()
 
-        return ScriptRunResult(response, self._variables)
+        return ScriptRunResult(self._conversation_response, response, self._variables)
```

Comment:
If I'm reading this right, the only variables used for templating in the response rendering are the current script variables, correct?

I feel that the way variables are handled in conversation responses is all over the place. It's not the same between built-in `intents` reponses vs `custom_sentences` vs `intent_script` responses vs this. Neither the same variables, nor the same approach is used throughout.

Reply:
Nevermind, I've read the docs. It's not the script variables, it's the manually defined variables. Still, it seems complicated to pass a variable to the response, and still very different than composing other responses in other ways.